File: ObjC/LightboxArrayController.m
//
//  LightboxArrayController.m
//  OCTestArrayBinding
//
//  Created by Steve Ross on 1/11/10.
//  Copyright 2010 Calico Web Development. All rights reserved.
//

#import "LightboxArrayController.h"
#import "Lightbox.h"

@implementation LightboxArrayController
- (void) awakeFromNib {
  NSLog(@"initializing");
  Lightbox *theLightbox = [[Lightbox alloc] init];
  theLightbox.title = @"Title 123";
  theLightbox.number = 112233;
  [self addObject:theLightbox];
  theLightbox.title = @"Title 234";
  theLightbox.number = 223344;
  [self addObject:theLightbox];
}
@end

File: MacRuby/LightboxArrayController.rb
# LightboxArrayController.rb
# IStockCocoa
#
# Created by Steve Ross on 1/10/10.
# Copyright 2010 Calico Web Development. All rights reserved.

class Lightbox
  attr_accessor :title
  
  def initialize(args = {})
    args.each_pair do |k, v|
      instance_variable_set("@#{k.to_s}", v)
    end
  end

  def number
    @number.to_i
  end
  
  def number=(v)
    @number = v.to_i
  end
  
  def to_s
    "number => #{self.number} (#{self.number.class}) title => #{self.title} (#{self.title.class})"
  end
end

class LightboxArrayController < NSArrayController
  def awakeFromNib
    [
      Lightbox.new(:number => 23456, :title => 'lightbox 123456'),
      Lightbox.new(:number => 134567, :title => 'lightbox 234567')
    ].each do |object|
      puts "adding #{object}"
      self.addObject(object)
    end
  end
end